dp/core/syscall.c

```diff
@@ -199,7 +199,8 @@ static int sys_bpoll(struct bsys_desc __user *d, unsigned int nr)
 				KSTATS_PUSH(idle, NULL);
 
 				start = rdtsc();
-				eth_rx_idle_wait(deadline);
+				if (percpu_get(eth_num_queues) > 0)
```

Comment:
I would follow the same pattern as `eth_process_poll`. That means that you should define a function in `ethqueue.c` and call it from here.

Also, we have to think how to deal with multiple NIC because this function can wait up to 10 msecs for an incoming packet, so it is acceptable to sequentially wait on all the NICs (up to 4).